package cfg_messages;


message Param {
    required string name = 1;
    required string value = 2; // value should be json encoded!
}


message ConfigParamsRequest {
    required string sender = 1;
    required string receiver = 2;
    repeated string param_names = 3;
}

message ConfigParams {
    required string sender = 1;
    optional string receiver = 2;
    repeated Param params = 3;
}

message PeerIdentity {
    required string peer_id = 1;
}

message PeerReadyQuery {
    required string sender = 1;
    repeated string deps = 2;
}

message PeerReadyStatus {
    required string receiver = 1;
    required bool peers_ready = 2;
}

message ConfigError {
    optional string rq_type = 1;
    optional string error_str = 2;
    optional string errno = 3;
}

message LauncherCommand {
    required string sender = 1;
    required string serialized_msg = 2; // serialized to json, templates in launcher_messages.py
}